src/main/java/org/opensearch/ad/cluster/ADVersionUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.cluster;
+
+import static org.opensearch.ad.constant.CommonName.AD_PLUGIN_VERSION_FOR_TEST;
+
+import org.opensearch.Version;
+
+public class ADVersionUtil {
+
+    public static final int VERSION_SEGMENTS = 3;
+
+    public static Version fromString(String adVersion) {
+        if (AD_PLUGIN_VERSION_FOR_TEST.equals(adVersion)) {
+            return Version.CURRENT;
+        }
+        return Version.fromString(normalizeVersion(adVersion));
+    }
+
+    public static String normalizeVersion(String adVersion) {
+        if (adVersion == null) {
+            throw new IllegalArgumentException("Wrong AD version " + adVersion);
```

Comment:
Minor: change the exception to logically match the fact that `adVersion` was null?

Reply:
sure, will tune error message